x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/results/AutodetectResult.java

```diff
@@ -165,6 +176,9 @@ public void writeTo(StreamOutput out) throws IOException {
         writeNullable(modelPlot, out);
         writeNullable(annotation, out);
         writeNullable(categoryDefinition, out);
+        if (out.getVersion().onOrAfter(Version.V_8_0_0)) {
+            writeNullable(categorizerStats, out);
```

Comment:
```suggestion
            out.writeOptionalWriteable(categorizerStats);
```

I wonder if this `writeNullable` method could just be removed? The down side is that there is no `StreamOutput#writeOptionalList`. Maybe there should be?